platform-sdk/docs/core/dynamic-address-book/dynamic-address-book.md

```diff
@@ -0,0 +1,183 @@
+# Dynamic AddressBook (DAB)
+
+This document details the initial top level design for dynamic address book and defines key concepts and properties
+that must be factored into the implementation of dynamic address book. This document will need to be updated to
+match any changes to the implementation. Adjacent documentation goes into more detail on the implementation of
+dynamic address book.
+
+The end goal of the dynamic address book effort is to allow the address book to be updated through `Hedera API (HAPI)
+transactions` without requiring a software upgrade or network restart.
+
+![Dynamic Address Book Design](./DynamicAddressBook-Design.drawio.svg)
+
+# History and Background
+
+Prior to the dynamic address book effort, the platform's address book was static during the execution of the nodes
+on the network. Adding a node or removing a node to the address book would require a software upgrade and network
+wide restart for the nodes to pick up a new address book through `config.txt`. Outside the maintenance window of
+performing a software upgrade, the address book was saved to the state and used on node restarts, ignoring `config.txt`.
+
+During the initial design phase of the dynamic address book effort, there was a push to merge the two concepts of
+address book, one in the application, and one in the platform. The final result was that the platform still needed
+its own separate data structure. The platform's address book has been renamed to `Consensus Roster` or `Roster`.
+The consensus roster tells the platform which nodes are creating events and participating in consensus and provides
+the necessary metadata from each node such as node weight, public keys, and gossip endpoints.
+
+The platform's `ConsensusRoster` and the application's `AddressBook` are deeply related but do not have the same
+lifecycle. Non-Platform impacting changes to the address book can take effect in the round when the HAPI
+transactions are handled. Because consensus happens in a pipeline, changes to the consensus roster cannot take
+effect the same round a new roster is created. At minimum the new roster must wait the number of rounds in the
+pipeline before it can take effect. If nodes are added or removed or keys are rotated, coordination between nodes
+for re-keying protocols may increase the number of rounds before a new consensus roster can take effect.
```

Comment:
The wording here needs massaging.

1. The consensus roster is derived from the address book. They are not conceptually different entities and it is important that we talk about it and convey it to the outside world that way. The roster is merely a view or subset of of the address book that the platform cares about.

2. I am not a fan of saying that the address book and the roster have different lifecycles. The address book is always updated immediately, including data that is included in the consensus roster view. The part that is not immediate is the platform adoption of those changes.